loadtest/user/userentity/actions.go

```diff
@@ -21,16 +21,17 @@ import (
 	"github.com/mattermost/mattermost/server/public/model"
 )
 
-// Possible actions for the openID authentication
-type authOpenIDAction string
+// Possible actions for the IDP authentication
+type authIDPAction string
 
 const (
-	authOpenIDLogin  authOpenIDAction = "login"
-	authOpenIDSignup authOpenIDAction = "signup"
+	authIDPLogin  authIDPAction = "login"
+	authIDPSignup authIDPAction = "signup"
 )
 
 var (
-	openIDLoginFormActionRegex = regexp.MustCompile(`action=["'](.*?)["']`)
+	keycloakIDPLoginFormActionRegex = regexp.MustCompile(`action=["'](.*?)["']`)
+	keycloakSAMLInputValueRegex     = regexp.MustCompile(`input type=["']hidden["'] name=["'](\w+)["'] value=["'](.*?)["']`)
```

Comment:
"a mere glimpse of the world of reg​ex parsers for HTML will ins​tantly transport a programmer's consciousness into a world of ceaseless screaming" https://stackoverflow.com/a/1732454/3248221

(It's good for this specific case, just kidding)